### PR TITLE
edn: allow comments and improve whitespace handling

### LINKED ISSUE #166: Test EDN parser against leading, trailing, and inter-element whitespace
I saw issues parsing ` [1]`, `[1] `, and `[ [1]]`.  These are simplified and from memory, so they may not all fail.  For what it's worth I was using `r#` raw strings, although I don't think that's relevant.

### PR DESCRIPTION
**Pull request by ncalexan**
@rnewman, @joewalker: review?

Fixes #166.

### CODE REVIEW

File: edn/src/edn.rustpeg [hunk 1]
```diff
@@ -105,37 +104,41 @@ keyword -> Value
...
 value -> Value
-    = nil / boolean / float / bigint / integer / text /
+    = __ v:(nil / boolean / float / bigint / integer / text /
       keyword / symbol /
-      list / vector / map / set
+      list / vector / map / set) __ {
+    v
+}
+
+whitespace = (" " / "\r" / "\n" / "\t" / ",")
```
Comment (rnewman) [inline edn/src/edn.rustpeg]: It's worth commenting that Clojure, and thus EDN, regard commas as whitespace, and thus this is not a mistake; `[5,,,,4]` is totally legit syntax for a two-element array.

(It's also worth noting that idiomatic EDN doesn't bother with commas.)

File: edn/src/edn.rustpeg [hunk 2]
```diff
@@ -105,37 +104,41 @@ keyword -> Value
...
+map -> Value = "{" v:(pair)* "}" {
     Value::Map(BTreeMap::from_iter(v))
 }
 
 // It's important that float comes before integer or the parser assumes that
 // floats are integers and fails to parse
 #[export]
 value -> Value
-    = nil / boolean / float / bigint / integer / text /
+    = __ v:(nil / boolean / float / bigint / integer / text /
```
Comment (joewalker) [inline edn/src/edn.rustpeg]: Yup. Like it much better than the blind alley we were obviously headed down.

File: edn/src/edn.rustpeg (hunk 1 above)
Comment (joewalker) [inline edn/src/edn.rustpeg]: Except presumably for maps, right - you need commas there?

Comment (rnewman) [inline edn/src/edn.rustpeg]: No — `{:foo 5 :bar 3}` is totally normal. One would only add commas if they helped clarity, as in `{5 3, 3 2, 2 3}`.

(Swift does the same thing for numbers, and Rust might too — `1_234_567` is acceptable numeric syntax, as is `12_34_56_78` if you live on bizarro planet.)

**Pull request merged**